phpunit/experimental/interactivity-api/directive-processing-test.php

```diff
@@ -76,8 +76,103 @@ public function test_directives_with_double_hyphen_processed_correctly() {
 		$tags   = new WP_HTML_Tag_Processor( $markup );
 		gutenberg_interactivity_process_directives( $tags, 'foo-', $directives );
 	}
+
+	public function test_interactivity_process_directives_in_root_blocks() {
+		// Reset root blocks counter.
+		WP_Directive_Processor::$root_blocks = array();
+		$provider                            = $this->data_only_root_blocks_are_processed();
+		foreach ( $provider as $provider ) {
+			$parsed_blocks = parse_blocks( $provider['page_content'] );
+			foreach ( $parsed_blocks as $parsed_block ) {
+				render_block( $parsed_block );
+			}
```

Comment:
Out of curiosity: what's the benefit of doing this over using `do_blocks` directly?

Reply:
Using `render_block` was also testing that `gutenberg_interactivity_process_directives_in_root_blocks` is also updating `$do_block_calls` as expected.

Now that we are going to use a different approach I can revisit this test.

Reply:
I meant, why not to use `do_blocks` to do the tests:

```suggestion
			do_blocks( $provider['page_content'] );
```

Reply:
That would do the job, true.

Reply:
Ok, thanks! 😄